engine/configurator/conf_cyaml.h

```diff
@@ -0,0 +1,297 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/** @file
+ * @brief YAML configuration types and schema
+ *
+ * API definitions.
+ *
+ * Copyright (C) 2018-2022 OKTET Labs Ltd. All rights reserved.
+ */
+
+#ifndef __TE_CONF_CYAML_H__
+#define __TE_CONF_CYAML_H__
+
+#include <cyaml/cyaml.h>
+#include "conf_api.h"
+#include "conf_types.h"
+#include "logger_api.h"
+
+/******************************************************************************
+ * C data structure for storing a backup.
+ *
+ * This is what we want to load the YAML into.
+ ******************************************************************************/
+
+/** Enumeration for object or instance. It answers if it is object */
+enum scope {
+    CFG_INSTANCE = 0, /* It is a default value */
+    CFG_OBJECT
+};
+
+/** Structure for storing depends entry */
+typedef struct depends_entry {
+    char *oid;
+    uint8_t scope; /* optional; 0 is default value means instance */
+} depends_entry;
+
+/**
+ * Structure for storing object information. The name for the field coincides
+ * with the corresponding YAML key except 'def_val' for 'default',
+ * 'volat' for 'volatile' and 'no_parent_dep' for 'parent_dep'. In the last
+ * case there is also a logical negation.
+ */
+typedef struct object_type {
+    char *d; /* cyaml doesn'know about this type of field */
+
+    char *oid;
+    uint8_t access;
+    uint8_t type; /* optional; 0 is default value means NONE type */
+    te_bool unit; /* optional; 0 is default value */
+    char *def_val; /* optional; NULL is default value */
+    te_bool volat; /* optional; 0 is default value */
+    te_bool substitution; /* optional; 0 is default value */
+    te_bool no_parent_dep; /* optional; 0 is default value */
```

Comment:
If you have Doxygen description for the structure, you should have proper Doxygen descriptions for members as well.

Reply:
OK, I've added them.